plugins/modules/slack.py

```diff
@@ -297,7 +297,7 @@ def build_payload_for_slack(text, channel, thread_id, username, icon_url, icon_e
         # With a custom color we have to set the message as attachment, and explicitly turn markdown parsing on for it.
         payload = dict(attachments=[dict(text=escape_quotes(text), color=color, mrkdwn_in=["text"])])
     if channel is not None:
-        if channel.startswith(('#', '@', 'C0', 'GF', 'G0', 'CP')):
+        if channel.startswith(('#', '@')) or re.match(r'^[C|D|G][A-Za-z0-9]{8,10}$', channel):
```

Comment:
Where do you know from that the length can be exactly 8 or 10, but no other value? I'm really skeptical that this regex is valid if this information is just based on three IDs that were found as examples in an API documentation.

Reply:
I agree. I think it would be better to just remove this channel value check which is only use to automatically append a #